File: admin-ms-flask/app/migrations/scripts/20250203120220-create-initial-schema.py
from pymongo import MongoClient
from app.extensions.mongodb import MongoConnection
from app.utils.file import FileUitl 
import os



def run_db_changes(db: MongoClient):
    pass



def migrate_up(conn : MongoConnection):
    script_file = os.path.join(os.path.dirname(__file__), "../schema-data/20250203120220-create-initial-schema.js")
    FileUitl.execute_mongo_script(script_file, conn)
    return

    # db = conn.getDb()
    # collections = db.list_collection_names()
    # if not "admin_users" in collections:
    #     validator = {
    #         "$jsonSchema": {
    #             "bsonType": "object",
    #             "required": ["email", "username"],
    #             "properties": {
    #                 "email": {
    #                     "bsonType": "string",
    #                     "description": "must be a string and is required"
    #                 },
    #                 "username": {
    #                     "bsonType": "string",
    #                     "description": "must be a string and is required"
    #                 }
    #             }
    #         }
    #     }

    #     db.command("create", "admin_users", validator=validator, validationLevel="strict", validationAction="error")
    #     db.admin_users.create_index([("email", 1)], unique=True)

    #  if not "acl_roles" in collections:
    #     validator = {
    #             "$jsonSchema" : {
    #                 "bsonType" : "object", 
    #                 "required" : ["role_name"],
    #                 "properties" : {
    #                     "role_name" : {
    #                         "bsonType" : "string",
    #                         "description" : "must be a string and is required"
    #                     }
    #                 }
    #         }
    #     }
    #     db.command("create", "acl_roles", validator=validator, validationLevel="strict", validationAction="error")
    #     db.acl_roles.create_index([("role_name", 1)], unique=True)


def migrate_down():
    pass

File: admin-ms-flask/app/migrations/schema-data/20250203120220-create-initial-schema.js


let admin_users_exists = db.getCollectionNames().includes('admin_users');
if(!admin_users_exists) {
    db.createCollection("admin_users", {
        validator : {
            $jsonSchema : {
                bsonType : "object", 
                required : ["email", "username"],
                properties : {
                    email : {
                        bsonType : "string",
                        description : "must be a string and is required"
                    },
                    username : {
                        bsonType : "string",
                        description : "must be an integer, value should be less than 18 and required"
                    }
                }
            }
        },
        validationLevel : "strict",
        validationAction : "error"     
     });
 
    db.getCollection('admin_users').createIndex({"email" : 1}, {unique: true});
}

let acl_role_exists = db.getCollectionNames().includes('acl_roles');

if(!acl_role_exists) {
    db.createCollection("acl_roles", {
        validator : {
            $jsonSchema : {
                bsonType : "object", 
                required : ["role_name"],
                properties : {
                    role_name : {
                        bsonType : "string",
                        description : "must be a string and is required"
                    }
                }
            }
        },
        validationLevel : "strict",
        validationAction : "error"     
    });
     
     
    db.getCollection('acl_roles').createIndex({"role_name" : 1}, {unique: true});
}
